packages/web-components/src/components/va-telephone/va-telephone.tsx

```diff
@@ -71,20 +128,45 @@ export class VaTelephone {
    * @return {string} - Combined phone number parts within the label seperated by
    * periods, e.g. "800-555-1212" becomes "8 0 0. 5 5 5. 1 2 1 2"
    */
-  static formatTelLabel(number: string): string {
-    return number
-      .split(/[^\d\w]+/)
-      .filter(n => n)
-      .map(chunk => (chunk === 'ext' ? 'extension' : chunk.split('').join(' ')))
-      .join('. ');
+  static formatTelLabel(number: string, vanity: boolean): string {
+    return vanity
+      ? number
+          .split(/[^\d]+/)
+          .filter(n => n)
+          .map(chunk => chunk.split('').join(' '))
+          .join('. ')
+      : number
+          .split(/[^\d\w]+/)
+          .filter(n => n)
+          .map(chunk =>
+            chunk === 'ext' ? 'extension' : chunk.split('').join(' '),
+          )
+          .join('. ');
   }
 
-  static createHref(contact: string, extension: number): string {
+  static createHref(
+    contact: string,
+    extension: number,
+    alphaPhoneNumber: Function,
+    vanity: boolean,
+  ): string {
     const isN11 = contact.length === 3;
     // extension format ";ext=" from RFC3966 https://tools.ietf.org/html/rfc3966#page-5
     // but it seems that using a comma to pause for 2 seconds might be a better
     // solution - see https://dsva.slack.com/archives/C8E985R32/p1589814301103200
-    const href = `tel:${isN11 ? contact : `+1${contact}`}`;
+    const last4 = contact.slice(contact.length - 4);
+    let vanityNumber: string;
+    if (vanity) {
+      vanityNumber =
+        contact.substring(0, contact.length - 4) +
+        last4
+          .split('')
+          .map(digit => alphaPhoneNumber(digit))
+          .join('');
+    }
+    const href = `tel:${
+      isN11 ? contact : vanity ? `+1${vanityNumber}` : `+1${contact}`
+    }`;
```

Comment:
This is a little hard for me to understand, and I may ask for some simplifications. But regardless, can we get [some unit tests added here](https://github.com/department-of-veterans-affairs/component-library/blob/3d956e06839b6bfddc1d00a64a222f3829209297/packages/web-components/src/components/va-telephone/test/va-telephone.spec.tsx) so that even if the function is tricky to understand, the tests describe what it can do, what output to expect from the input, etc.